src/test/rbf_tests.cpp

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#include <policy/policy.h>
+#include <policy/rbf.h>
+#include <txmempool.h>
+#include <util/system.h>
+#include <util/time.h>
+
+#include <test/util/setup_common.h>
+
+#include <boost/test/unit_test.hpp>
+#include <optional>
+#include <vector>
+
+BOOST_FIXTURE_TEST_SUITE(rbf_tests, TestingSetup)
+
+inline CTransactionRef make_tx(std::vector<CAmount>&& output_values,
+                               std::vector<CTransactionRef>&& inputs=std::vector<CTransactionRef>(),
+                               std::vector<uint32_t>&& input_indices=std::vector<uint32_t>())
+{
+    CMutableTransaction tx = CMutableTransaction();
+    tx.vin.resize(inputs.size());
+    tx.vout.resize(output_values.size());
+    for (size_t i = 0; i < inputs.size(); ++i) {
+        tx.vin[i].prevout.hash = inputs[i]->GetHash();
+        tx.vin[i].prevout.n = input_indices.size() > i ? input_indices[i] : 0;
```

Comment:
nit:

```suggestion
        tx.vin[i].prevout.n = input_indices[i];
```

I'm not sure it's worth having this mechanism of fallback to `0` but only for the last inputs, it's a bit confusing.
Maybe just making `input_indices` completely optional (in which case you'd use `0` for all inputs) is better?
And if it's provided then it should be provided for all inputs.
In that case it would be:

```suggestion
        tx.vin[i].prevout.n = input_indices.size() > 0 ? input_indices[i] : 0;
```

Feel free to ignore though if you disagree.